text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,517 @@
+be mutually exclusive with _all_ elements of "B" (The inverse is also true).
+
+So each element of "A" is compared against each element of "B" using the following rules:
+- A `target-triple` is mutually exclusive with another `target-triple` if they are different.
+- A `target-triple` is mutually exclusive with a `cfg(..)` if the `cfg(..)` is not satisfied by
+    the `target-triple`.
+- A `cfg(all(A, B, ...))` is mutually exclusive with a `cfg(all(C, D, ...))` if any element of the list
+    `A, B, ...` is mutually exclusive with any element of the list `C, D, ...`.
+
+_Note_: All possible cases have been covered, since `cfg(A) == cfg(all(A))`.
+
+Two `cfg` singletons are mutually exclusive under the following rules:
+- `cfg(A)` is mutually exclusive with `cfg(not(A))`.
+- `cfg(<option> = "A")` is mutually exclusive with `cfg(<option> = "B")` if `A` and `B` are different,
+    and `<option>` has mutually exclusive elements.
+
+Some `cfg` options have mutually exclusive elements, while some do not. What is meant here is, for example,
+`target_arch = "x86_64"` and `target_arch = "arm"` are mutually exclusive (a target-triple cannot have both),
+while `target_feature = "avx"` and `target_feature = "rdrand"` are not.
+
+`cfg` options that have mutually exclusive elements:
+- `target_arch`
+- `target_os`
+- `target_env`
+- `target_abi`
+- `target_endian`
+- `target_pointer_width`
+- `target_vendor`
+
+Those that do not:
+- `target_feature`
+- `target_has_atomic`
+- `target_family`
+
+`target_family = "windows` and `target_family = "unix"` could also be defined as mutually exclusive to
+enhance usability, but this is left as a [future possibility](#more-cfg-relations).
+
+## Behavior with unknown entries (and custom targets)
+
+Just as `[target.my-custom-target.dependencies]` is allowed by `cargo`, `supported-targets` can contain
+unknown entries. This is important because users may have different `rustc` versions, and the set of
+official `rustc` targets is unstable; Targets can change name or be removed. Also, developers 
+may want to support their custom target.
+
+To determine if an entry in `supported-targets` is a target name or a `cfg` specification, the
+same mechanism as for `[target.'cfg(..)']` is used (using
+[`cargo-platform`](https://docs.rs/cargo-platform/latest/cargo_platform/index.html)).
+
+## Ignoring builds for unsupported targets
+
+When the target used is not supported by the package being built, the package will either be
+skipped or an error will be raised, depending on how `cargo` was invoked. If cargo is invoked
+in a workspace or virtual workspace without specifying a specific package, then `cargo` skips the package.
+If a specific package is specified using `--package`, or if `cargo` is invoked on a single package,
+then an error is raised.
+
+## Eliminating unused dependencies from `Cargo.lock`
+
+A package's dependencies may themselves have `[target.'cfg(..)'.dependencies]` tables, which may never be used because of the
```

Comment:
As this affects dependency resolution, we'll also need to discuss the relevant changes to the Index.

For the existing schema, see https://doc.rust-lang.org/cargo/reference/registry-index.html#json-schema